File: python/rna-transcription/rna_transcription.py
def to_rna(dna_strand):
  dna_to_rna = {
  	'G':'C',
    'C':'G',
    'T':'A',
    'A':'U'
    }
  try:
    # lookup key, put values into a list, then convert to string
    rna_strand = ''.join([ dna_to_rna[x] for x in dna_strand.upper() ])
  except KeyError:
    raise ValueError("Invalid input: DNA strand must consist of only GCTA")
  return rna_strand

# use dictionary key 
# exception handling

File: javascript/rna-transcription/rna-transcription.js
var DNATranscriber = function () {
}

DNATranscriber.prototype.toRna = function(DNAstrand){
    const PairMap = {
  	'G':'C',
    'C':'G',
    'T':'A',
    'A':'U'
    }
    
    return DNAstrand.split('').map(x => {
        if (PairMap[x]){
            return PairMap[x];
        }
        else {
            throw new Error('Invalid input')
        }
    }).join('')
    
};

module.exports = DNATranscriber;